lib/peer-pool.js

```diff
@@ -118,7 +118,12 @@ class PeerPool {
   send (peerId, message) {
     const peerConnection = this.peerConnectionsById.get(peerId)
     if (peerConnection) {
-      peerConnection.send(message)
+      try {
+        peerConnection.send(message)
+      } catch (error) {
+        peerConnection.disconnect()
+      }
```

Comment:
@CaptainJohnyAppleSeed: Thanks for sharing the steps in https://github.com/atom/teletype-client/pull/60#issuecomment-381710207. I ran through those steps, and I verified that this change prevents the RTCDataChannel error. :bow:

---

@as-cii: Can you help me figure out whether this change is the preferred way to resolve this issue?

When following the steps in https://github.com/atom/teletype-client/pull/60#issuecomment-381710207, the [call to `RTCDataChannel::send()`](https://github.com/atom/teletype-client/blob/f0cf0054d6ba3e923a31860ceefd0d78b94cdf99/lib/peer-connection.js#L235) throws a [NetworkError](https://developer.mozilla.org/en-US/docs/Web/API/RTCDataChannel/send#Exceptions). `PeerConnection::send()` then [rethrows that error](https://github.com/atom/teletype-client/blob/f0cf0054d6ba3e923a31860ceefd0d78b94cdf99/lib/peer-connection.js#L238). With the current changes in this pull request, we catch any error thrown by `PeerConnection::send()`, and we disconnect from the peer. That disconnection event propagates from the `PeerConnection` to the `PeerPool` to the `StarOverlayNetwork` to the `Portal`.

This pull request resolves the problem described in https://github.com/atom/teletype-client/pull/60#issuecomment-381710207, but I have a few questions:

1. The current implementation in this pull request disconnects for [any error type that can occur in `RTCDataChannel::send()`](https://developer.mozilla.org/en-US/docs/Web/API/RTCDataChannel/send#Exceptions). Is that reasonable, or would it make sense to only disconnect when a `NetworkError` occurs?
2. Instead of having `PeerPool` catch this exception, should we change `PeerConnection` to catch the `NetworkError` and disconnect [like we do in `PeerConnection::receiveSignal()`](https://github.com/atom/teletype-client/blob/f0cf0054d6ba3e923a31860ceefd0d78b94cdf99/lib/peer-connection.js#L189-L190)?

    In `PeerConnection::receiveSignal()`, we also [call `PeerConnection::handleError()`](https://github.com/atom/teletype-client/blob/f0cf0054d6ba3e923a31860ceefd0d78b94cdf99/lib/peer-connection.js#L191), which [notifies observers of the error](https://github.com/atom/teletype-client/blob/f0cf0054d6ba3e923a31860ceefd0d78b94cdf99/lib/peer-connection.js#L171). And [one of those observers is the `PeerPool`](https://github.com/atom/teletype-client/blob/v0.37.2/lib/peer-pool.js#L164-L167). I wonder if a similar approach is appropriate in `PeerConnection::send()`.